engine/src/main.cpp

```diff
@@ -384,7 +384,8 @@ int main( int argc, char *argv[] )
         //Specify the config file and the possible mod subdir to play
         VSFileSystem::InitPaths( CONFIGFILE, subdir );
     }
-
+    Configuration postConfigLoad;
+    memcpy(&configuration, &postConfigLoad, sizeof(configuration)); // should define an = operator as this, but, eh - just for testing.
```

Comment:
No `memcpy`. Bad `memcpy`. Not for complex types like structs and classes, at any rate.

The = operator sounds like a better idea.

Reply:
Yup. Quick and dirty 3am code is ... exactly-as-awful-as-expected 3am code. I don't actually _want_ this accepted before I fix this, but I also don't want to delay the rest of this PR being reviewed/tested any further. I expect to be on baby duty for much of the day, so I'll try to get to this to tonight/tomorrow. This is known-flawed-and-dangerous for non-POD structs, but it enables testing of the config functionality, etc. 

Re = operator -- that will work for now and be generally potentially useful indefinitely. Longer term, I think the discussion was moving towards upgrading Configuration to have methods, while keeping its members as simple structs, and (re)init/setters would then be methods in Configuration. If we want it to initialize at construction, Configuration can only init to defaults, and we should be explicit in documenting this in the code and using obvious patterns (i.e. If construction is default, just use the default, not the call that looks like it's getting a value, and to keep default specification singular, the XML reader function used to init should then not have a default field at all)